### ISSUE TITLE
Provide an interface for `EIP712`

### ISSUE THREAD

**Issue by PaulRBerg** [Uniswap/permit2#193]
There is a problem with the current contract architecture that makes it difficult to write tests that need access to Permit2's [`DOMAIN_SEPARATOR`](https://github.com/Uniswap/permit2/blob/db96e06278b78123970183d28f502217bef156f4/src/EIP712.sol#L24).

Unless the integrating project uses the same Solidity compiler version of Permit2 (i.e. v0.8.17), the `AllowanceTransfer` contract cannot be imported. Only the interfaces can (e.g. `IAllowanceTransfer`). The implicit assumption is (as I understand it) that users are meant to use the [`DeployPermit2`](https://github.com/Uniswap/permit2/blob/db96e06278b78123970183d28f502217bef156f4/test/utils/DeployPermit2.sol) script, and cast the returned address as `IAllowanceTransfer`.

Because the `IAllowanceTransfer` interface does not contain a function for accessing `DOMAIN_SEPARATOR`, users must resort to doing something like this:

```solidity
import { DeployPermit2 } from "permit2/test/utils/DeployPermit2.sol";

interface IEIP712 {
    function DOMAIN_SEPARATOR() external view returns (bytes32);
}

contract MyTest is DeployPermit2 {
     IAllowanceTransfer permit2;

    function testSomething() {
        permit2 = IAllowanceTransfer(deployPermit2());
        bytes32 domainSeparator = IEIP712(address(permit2));
    }
}
```

It would be much easier if Permit2 had an `IEIP712` interface, inherited in both `IAllowanceTransfer` and `ISignatureTransfer`.

**Comment by PaulRBerg** [comment]
We wrote an implementation for this `IEIP712` interface in our fork of Permit2:

https://github.com/sablierhq/permit2/commit/02f824ab187fe356d389ab14bda9717a8be06bfd

**Comment by PaulRBerg** [comment]
Closed via #215.
